### ISSUE TITLE
Ability to skip type checking of remote modules

### ISSUE THREAD

**Issue by dsherret** [denoland/deno#5460]
The TS compiler has a [`--skipLibCheck`](https://www.typescriptlang.org/docs/handbook/compiler-options.html) compiler option that allows skipping type checking of declaration files.

Similarly, it might be nice for Deno to:

1. Have a way to not type check the private implementations of remote modules or make this the default (this would help with compile times too).
2. Have something like the `--skipLibCheck` compiler option that skips type checking of remote modules completely or have a way to do this on a case-by case basis. Example, in *deps.ts* people could add a comment above the import:

    ```
    // @ts-skip-check
    export * as myProject from "https://deno.land/x/my-project/etc....";
    ````

The reason why this would be nice is because remote modules are often out of our control and we might want to depend on an old version of a dependency that now has type checking errors in the current version of the TS compiler being used.

**Comment by nayeemrmn** [comment]
I suggest something like:
```
-t, --type-check=[all|local|none|<whitelist>...]
        Which modules should be type checked (default: local)
```
Ideally that would be a blacklist instead of a whitelist if there were some syntactic convention for that.

**Comment by bartlomieju** [comment]
@kitsonk what's your take on this issue?

**Comment by kitsonk** [comment]
It seems like a lot of work which would cause a level of confusion.  I think we should land the binary capability to skip type checking and live with that for a while before making the whole thing more complicated.

**Comment by bartlomieju** [comment]
> It seems like a lot of work which would cause a level of confusion. I think we should land the binary capability to skip type checking and live with that for a while before making the whole thing more complicated.

So `--no-type-check` flag that has global effect 👍 makes sense for a first pass

**Comment by bartlomieju** [comment]
`--no-check` option landed in #6456, will be released in `v1.2.0`
